From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alastair D'Silva <alastair@d-silva.org>
Date: Fri, 18 Sep 2026 13:28:00 +1000
Subject: include: configs: sunxi-common: prefer SD card over eMMC and NVMe

Ensure that mmc0 (MicroSD card) is always scanned and booted before
mmc1 (eMMC) and NVMe, even if U-Boot was loaded from eMMC.

Also update mmc_bootdev prior to executing bootcmd_mmc0 / bootcmd_mmc1
so that downstream boot scripts (such as boot-sunxi.cmd) that rely on
${mmc_bootdev} load the kernel, dtb, and environment from the selected
MMC device.

Signed-off-by: Alastair D'Silva <alastair@d-silva.org>
---
 include/configs/sunxi-common.h | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

--- a/include/configs/sunxi-common.h
+++ b/include/configs/sunxi-common.h
@@ -165,13 +165,10 @@
 	BOOTENV_DEV_MMC(MMC, mmc, 0)					\
 	BOOTENV_DEV_MMC(MMC, mmc, 1)					\
 	"bootcmd_mmc_auto="						\
-		"if test ${mmc_bootdev} -eq 1; then "			\
-			"run bootcmd_mmc1; "				\
-			"run bootcmd_mmc0; "				\
-		"elif test ${mmc_bootdev} -eq 0; then "			\
-			"run bootcmd_mmc0; "				\
-			"run bootcmd_mmc1; "				\
-		"fi\0"
+		"setenv mmc_bootdev 0; "				\
+		"run bootcmd_mmc0; "					\
+		"setenv mmc_bootdev 1; "				\
+		"run bootcmd_mmc1;\0"
 
 #define BOOTENV_DEV_NAME_MMC_AUTO(devtypeu, devtypel, instance) \
 	"mmc_auto "
-- 
Armbian
